homeassistant/helpers/entity.py

```diff
@@ -1048,16 +1066,16 @@ async def async_internal_added_to_hass(self) -> None:
 
         Not to be extended by integrations.
         """
-        info = {
+        info: EntityInfo = {
             "domain": self.platform.platform_name,
             "custom_component": "custom_components" in type(self).__module__,
+            "source": SOURCE_CONFIG_ENTRY
+            if self.platform.config_entry
```

Comment:
Can we avoid ternary operator?

Reply:
The "source" is only used by frontend, now removed from the dict and instead generated when sending the WS response